arrow-array/src/array/binary_array.rs

```diff
@@ -845,4 +852,22 @@ mod tests {
             .validate_full()
             .expect("All null array has valid array data");
     }
+
+    #[test]
+    fn test_empty_offsets() {
+        let string = BinaryArray::from(
+            ArrayData::builder(DataType::Binary)
+                .buffers(vec![Buffer::from(&[]), Buffer::from(&[])])
+                .build()
+                .unwrap(),
+        );
+        assert_eq!(string.value_offsets(), &[0]);
+        let string = LargeBinaryArray::from(
```

Comment:
```suggestion
        let string = LargeBinaryArray::from(
        assert_eq!(string.len(), 0);
```

I suggest you add a test for `len()` as described in the original bug report.

This comment applies to all the tests added in this PR